core/trie/snap_support_test.go

```diff
@@ -0,0 +1,285 @@
+			startQuery:       numToFelt(500),
+			expectedKeyCount: 5,
+			hasMore:          false,
+		},
+		{
+			name:             "start with limit query",
+			startQuery:       numToFelt(100),
+			limitQuery:       numToFelt(500),
+			expectedKeyCount: 5,
+			hasMore:          true,
+		},
+		{
+			name:             "start with limit query and node count limit",
+			startQuery:       numToFelt(100),
+			limitQuery:       numToFelt(500),
+			maxNode:          3,
+			expectedKeyCount: 3,
+			hasMore:          true,
+		},
+		{
+			name:             "finished before limit query",
+			startQuery:       numToFelt(100),
+			limitQuery:       numToFelt(20000),
+			expectedKeyCount: 9,
+			hasMore:          false,
+		},
+		{
+			name:             "last one right after limit query",
+			startQuery:       numToFelt(100),
+			limitQuery:       numToFelt(900),
+			expectedKeyCount: 9,
+			hasMore:          false,
+		},
+		{
+			name:             "two leaf after limit query, last leaf skipped",
+			startQuery:       numToFelt(100),
+			limitQuery:       numToFelt(800),
+			expectedKeyCount: 8,
+			hasMore:          true,
+		},
+		{
+			name:             "no node between start and limit",
+			startQuery:       numToFelt(450),
+			limitQuery:       numToFelt(451),
+			expectedKeyCount: 1,
+			hasMore:          true,
+		},
+		{
+			name:             "start query after last node",
+			startQuery:       numToFelt(10000),
+			expectedKeyCount: 0,
+			hasMore:          false,
+		},
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.name, func(t *testing.T) {
+			storage := trie.NewStorage(db.NewMemTransaction(), []byte{})
+			testTrie, err := trie.NewTriePedersen(storage, 251)
+			assert.NoError(t, err)
+
+			for i := 0; i < 10; i++ {
+				_, err = testTrie.Put(numToFelt(i*100+1), numToFelt(i*100+2))
+				assert.NoError(t, err)
+			}
+
+			expectedRoot, err := testTrie.Root()
+			assert.NoError(t, err)
+
+			startQuery := scenario.startQuery
+			var keys []*felt.Felt
+			var values []*felt.Felt
+
+			proofs, _, err := testTrie.IterateAndGenerateProof(startQuery, func(key, value *felt.Felt) (bool, error) {
+				keys = append(keys, key)
+				values = append(values, value)
+				if scenario.maxNode > 0 && len(keys) >= scenario.maxNode {
+					return false, nil
+				}
+				if scenario.limitQuery != nil && key.Cmp(scenario.limitQuery) > 0 {
+					// Last (one after limit) is included.
+					return false, nil
+				}
+				return true, nil
+			})
+			assert.NoError(t, err)
+
+			assert.Equal(t, scenario.expectedKeyCount, len(keys))
+			if scenario.noProof {
+				assert.Empty(t, proofs)
+			}
+
+			hasMore, valid, err := trie.VerifyRange(expectedRoot, startQuery, keys, values, proofs, crypto.Pedersen, trieHeight)
+			assert.NoError(t, err)
+			assert.True(t, valid)
+
+			assert.Equal(t, scenario.hasMore, hasMore)
+		})
+	}
+}
+
+func TestRangeAndVerifyReject(t *testing.T) {
+	scenarios := []struct {
+		name       string
+		startQuery *felt.Felt
+		skip       bool
```

Comment:
Why do we skip tests?

Reply:
Because it will fail if its not skipped.